utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,244 @@
+          connectivityImageView.setImage(connectivityImage.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided connectivitySymbol" + e.getMessage()).show();
+        }
+      });
+  
+      // Create a renderer for the associations
+      UniqueValueRenderer.UniqueValue attachmentUniqueValue  = new UniqueValueRenderer.UniqueValue("Attachment", "", attachmentSymbol, Arrays.asList(UtilityAssociationType.ATTACHMENT.toString()));
+      UniqueValueRenderer.UniqueValue connectivityUniqueValue  = new UniqueValueRenderer.UniqueValue("Connectivity", "", connectivitySymbol, Arrays.asList(UtilityAssociationType.CONNECTIVITY.toString()));
+  
+      // create a graphics overlay for associations
+      associationsOverlay = new GraphicsOverlay();
+      associationsOverlay.setRenderer(new UniqueValueRenderer(Arrays.asList("AssociationType"), Arrays.asList(attachmentUniqueValue, connectivityUniqueValue), "", null));
+  
+      // create a map with the topographic vector basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographicVector());
+      
+      // create a viewpoint to focus on the utility networks'extent
+      Viewpoint viewPoint = new Viewpoint(41.8057655, -88.1489692, 23);
+  
+      // set initial arcgis map extent
+      map.setInitialViewpoint(viewPoint);
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+      mapView.getGraphicsOverlays().add(associationsOverlay);
+      
+      // create the utility network
+      utilityNetwork = new UtilityNetwork("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer");
+  
+      // load utility network async and get all of the edges and junctions in the network
+      utilityNetwork.loadAsync();
+      utilityNetwork.addDoneLoadingListener(() -> {
+        List<UtilityNetworkSource> networkSources = utilityNetwork.getDefinition().getNetworkSources();
+    
+        for (UtilityNetworkSource nSource : networkSources)
+          // check an add all edges that are not subnet lines to the map.
+          if (nSource.getSourceType() == UtilityNetworkSource.Type.EDGE && nSource.getSourceUsageType() != UtilityNetworkSource.UsageType.SUBNET_LINE) {
+            mapView.getMap().getOperationalLayers().add(new FeatureLayer(nSource.getFeatureTable()));
+          }
+          // add all junctions to the map
+          else if (nSource.getSourceType() == UtilityNetworkSource.Type.JUNCTION) {
+            mapView.getMap().getOperationalLayers().add(new FeatureLayer(nSource.getFeatureTable()));
+          }
+    
+        // add association graphics at the initial view point
+        addAssociationsGraphics();
+        // listen for navigation changes
+        mapView.addNavigationChangedListener((event) -> {
+          addAssociationsGraphics();
+        });
```

Comment:
This can be refactored a little:
```suggestion
        mapView.addNavigationChangedListener(event -> addAssociationsGraphics());
```